respy/pre_processing/transition_matrix_exog_process.py

```diff
@@ -0,0 +1,169 @@
+        the exogenous process.
+    """
+    # Create categories for the different states of the process
+    categories = [
+        f"exogenous_process_{process_name}_{state}" for state in process_states
+    ]
+    index = pd.MultiIndex.from_tuples(
+        itertools.product(categories, states), names=["category", "name"]
+    )
+    params = pd.DataFrame(index=index, columns=["value"])
+    # Transfer values
+    for process_state in process_states:
+        params.loc[
+            (f"exogenous_process_{process_name}_{process_state}", states), "value"
+        ] = transformed_matrix[process_state].values
+    return params
+
+
+def transform_matrix(matrix):
+    """Transform probabilities to logit coefficients.
+
+    Parameters
+    ----------
+    matrix : pandas.DataFrame
+        The transition matrix for an exogenous process.
+    Returns
+    -------
+    transformed_matrix : pandas.DataFrame
+        The original DataFrame converted to logit values.
+
+    """
+    transformed_matrix = matrix.copy()
+    # Asign large negative numbers for zero probabilities
+    transformed_matrix[transformed_matrix == 0] = -1e300
+    transformed_matrix[transformed_matrix > 0] = np.log(
+        transformed_matrix[transformed_matrix > 0]
+    )
+    transformed_matrix.index = transformed_matrix.index.map(str)
+    return transformed_matrix
+
+
+def create_covariates(states, process_name, process_states):
+    """Create a covariate template for the user.
+
+    Parameters
+    ----------
+    states : list
+        A list of the states on which probabilities for the exogenous process are
+        definded.
+    process_name : string
+        The name of the exogenous process.
+    process_states : list
+        A list of the possible outcomes of the exogenous process.
+
+    Returns
+    -------
+    covariates : dictionary
+        A template for the specification of the covariates required for the exogenous
+        process.
+
+    """
+    covariates = {}
+    for state in states:
+        i = 0
+        for process_state in process_states:
+            if process_state in state:
+                destination_state = process_state
+                i += 1
+        # If state is independent of exogenous process, return without any specification
+        if i == 0:
+            covariates[str(state)] = "?"
+        elif i == 1:
+            # If only dependent on exogenous process, return full covariate statement
+            if state in process_states:
+                covariates[str(state)] = f"{process_name} == {destination_state}"
+            # If dependent on exogenous process and covariate, return template
+            else:
+                covariates[str(state)] = f"{process_name} == {destination_state} & ?"
+        else:
+            raise ValueError(
+                f"{state} contains more than one states of the exogenous " f"process."
+            )
+    return covariates
+
+
+def check_numerics(matrix):
```

Comment:
Private functions should start with "_". Unless this should be used outside of the module?

Reply:
Done!